presto-client/src/main/java/io/prestosql/client/ClientTypeSignature.java

```diff
@@ -87,6 +90,10 @@ public String toString()
             return rowToString();
         }
 
+        if (rawType.equals(VARCHAR) && arguments.get(0).getKind() == ParameterKind.LONG && arguments.get(0).getLongLiteral() == VARCHAR_MAX_LENGTH) {
+            return "varchar";
```

Comment:
Shouldn't the condition be `arguments.get(0).getLongLiteral() == Integer.MAX_VALUE`  (UNBOUNDED_LENGTH) rather than `... = Integer.MAX_VALUE - 1` (MAX_LENGTH) ?

https://github.com/prestosql/presto/blob/d3a1d8ca12b84524cd2d6f88fceb9a9c53df9700/presto-spi/src/main/java/io/prestosql/spi/type/VarcharType.java#L32-L34

Reply:
You are right, thanks!